yt_dlp/extractor/cinetecamilano.py

```diff
@@ -0,0 +1,56 @@
+from json import JSONDecodeError
+from urllib.error import HTTPError
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError, float_or_none, parse_iso8601, strip_or_none, traverse_obj,
+    urljoin)
+
+
+class CinetecaMilanoIE(InfoExtractor):
+    _VALID_URL = r"https?://(?:www\.)?cinetecamilano\.it/film/(?P<id>\d+)/?"
+    _TEST = {
+        "url": "https://www.cinetecamilano.it/film/1942",
+        "only_matching": True,
+    }
+
+    def _update_api_headers(self, headers):
+        token = self._get_cookies("https://www.cinetecamilano.it").get("cnt-token")
+        if token:
+            headers["Authorization"] = f"Bearer {token.value}"
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        headers = {"Referer": url}
+        self._update_api_headers(headers)
+        try:
+            film_json = self._download_json(
+                "https://www.cinetecamilano.it/api/catalogo/" + video_id,
+                video_id, headers=headers)
+        except ExtractorError as e:
+            if (isinstance(e.cause, HTTPError) and e.cause.code == 500) or (
+                isinstance(e.cause, JSONDecodeError)
+            ):
+                raise ExtractorError(
+                    "This video is only available for registered users. You may want to use --cookies.",
+                    expected=True, cause=e.cause,
+                )
+            else:
+                raise e
+        if not film_json.get("success") or not film_json["success"] or not film_json.get("archive"):
```

Comment:
If `success` does not exist this will fail, is that what you were wanting (not checking for `false`?)?

also this does the same thing
```suggestion
        if not film_json.get('success') or not film_json.get('archive'):
```